logger/src/lib.rs

```diff
@@ -686,86 +598,62 @@ impl Logger {
 
         set_max_level(Level::Trace.to_level_filter());
 
+        STATE.store(INITIALIZED, Ordering::SeqCst);
         self.log_helper(
             format!("Running {} v{}", app_info.name, app_info.version),
-            Some(Destination::Pipe),
+            Level::Info,
         );
-        LOGGER.level_info.set_writer(Destination::Pipe);
-        STATE.store(INITIALIZED, Ordering::SeqCst);
 
         Ok(())
     }
 
     // In a future PR we'll update the way things are written to the selected destination to avoid
     // the creation and allocation of unnecessary intermediate Strings. The log_helper method takes
     // care of the common logic involved in both writing regular log messages, and dumping metrics.
-    fn log_helper(&self, msg: String, maybe_forced_destination: Option<Destination>) {
-        let destination = maybe_forced_destination
-            .map(|forced_destination| forced_destination as usize)
-            .unwrap_or_else(|| self.level_info.writer());
-
-        // We have the awkward IF's for now because we can't use just "<enum_variant> as usize
-        // on the left side of a match arm for some reason.
-        match destination {
-            x if x == Destination::Pipe as usize => {
-                // Unwrap is safe cause the Destination is a Pipe.
-                if log_to_fifo(
-                    msg,
-                    self.log_fifo_guard()
-                        .as_mut()
-                        .expect("Failed to write to fifo due to poisoned lock"),
-                )
-                .is_err()
-                {
+    fn log_helper(&self, msg: String, msg_level: Level) {
+        if STATE.load(Ordering::SeqCst) == INITIALIZED {
+            if let Some(guard) = self.log_buf_guard().as_mut() {
+                if write_to_destination(msg, guard).is_err() {
                     // No reason to log the error to stderr here, just increment the metric.
                     METRICS.logger.missed_log_count.inc();
                 }
+            } else {
+                panic!("Failed to write to the provided metrics destination due to poisoned lock");
             }
-            x if x == Destination::Stderr as usize => {
-                eprintln!("{}", msg);
-            }
-            x if x == Destination::Stdout as usize => {
-                println!("{}", msg);
-            }
-            // This is hit on major program logic error.
-            _ => panic!("Invalid logger.level_info.writer!"),
+        } else if msg_level <= Level::Warn {
+            eprintln!("{}", msg);
+        } else {
+            println!("{}", msg);
         }
     }
 
-    /// Flushes metrics to the FIFO provided as argument upon initialization of the logger.
+    /// Flushes metrics to the destination provided as argument upon initialization of the logger.
     ///
```

Comment:
Nit: please erasee the empty comment line.

Reply:
Also, I would add a comment to elucidate the meaning of the return value.

Reply:
Done, take a look.